File: data/src/main/java/com/ykb/app/cryptotrader/data/dao/RoleDao.java
package com.ykb.app.cryptotrader.data.dao;

import com.ykb.app.cryptotrader.data.model.Role;
import com.ykb.app.cryptotrader.data.repo.RoleRepo;
import org.springframework.stereotype.Component;

@Component
public final class RoleDao extends Dao<Role, String> {

    public RoleDao(RoleRepo repo) {
        super(repo);
    }

}

File: library/data/data-auth/src/main/kotlin/com/ykb/app/cryptotrader/data/auth/dao/RoleDao.kt
package com.ykb.app.cryptotrader.data.auth.dao

import com.ykb.app.cryptotrader.data.auth.model.Role
import com.ykb.app.cryptotrader.data.auth.repo.RoleRepo
import com.ykb.app.cryptotrader.data.base.Dao
import org.springframework.stereotype.Service

@Service
class RoleDao(repo: RoleRepo) : Dao<Role, String>(repo)